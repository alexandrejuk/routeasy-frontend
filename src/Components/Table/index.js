import React from 'react'
import PropTypes from 'prop-types'

import styles from './style.module.css'

const Table = ({
  actionTable,
  data,
}) => {
  const renderRow = (row, index) => (
    <tr className={styles.tr} key={index}>
      <td className={styles.td}>{row.name}</td>
      <td className={styles.td}>{row.street}</td>
      <td className={styles.td}>{row.city}</td>
      <td className={styles.td}>{row.country}</td>
      <td className={styles.td}>{row.weight}</td>
      <td className={styles.td}>{row.lat}</td>
      <td className={styles.td}>{row.lng}</td>
      { actionTable && (
        <td className={styles.td}>
          <button
            data-testid="button"
            onClick={() => actionTable.control(row)}
          >
            {actionTable.label || 'Action'}
          </button>
        </td>
      )}
    </tr>
  )

  return (
    <table className={styles.table}>
      <thead>
        <tr className={styles.tr}>
          <th className={styles.th}>Nome</th>
          <th className={styles.th}>Rua</th>
          <th className={styles.th}>Cidade</th>
          <th className={styles.th}>Pa√≠s</th>
          <th className={styles.th}>Peso</th>
          <th className={styles.th}>Lat</th>
          <th className={styles.th}>Lng</th>
          {actionTable && (
            <th className={styles.th}>{actionTable.field || 'Action'}</th>
          )}
        </tr>
      </thead>
      <tbody>
        {
          data && data.length > 0
            ? data.map(renderRow)
            : <tr><td>No results found!</td></tr>
        }
      </tbody>
    </table>
  )
}

Table.propTypes = {
  actionTable: PropTypes.shape({
    control: PropTypes.func,
    field: PropTypes.string,
    label: PropTypes.string,
  }),
  data: PropTypes.arrayOf(
    PropTypes.shape({}).isRequired,
  ).isRequired,
}

Table.propTypes = {
  actionTable: null,
}

export default Table
