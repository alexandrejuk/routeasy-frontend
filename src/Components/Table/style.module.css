@import '../../assets/css/colors.css';

.table {
  border-collapse: collapse;
  font-family: arial, sans-serif;
  width: 100%;
}

.th {
  background-color: var(--silver-chalice);
  color: var(--white);
  text-align: center;
}

.td, .th {
  border: 1px solid var(--white);
  padding: 8px;
  text-align: left;
}

.tr:nth-child(even) {
  background-color: var(--mercury);
}

.tr:nth-child(odd) {
  background-color: var(--galery);
}
