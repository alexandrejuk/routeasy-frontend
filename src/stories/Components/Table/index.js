import React from 'react'
import { Table } from '../../../Components'
import { action } from '@storybook/addon-actions'

export default {
  title: 'Components/Table',
  component: Table,
}

const tableData = [{
  name: 'Eduardo',
  street: 'R. Ática',
  city: 'São Paulo',
  country: 'Brasil',
  lat: -23.531,
  lng: -46.584,
  weight: 10,
}]

const Template = ({
  actionTable,
  data,
}) => (
  <Table
    actionTable={actionTable}
    data={data}
  />
)

export const Default = Template.bind({})

Default.args = {
  actionTable: null,
  data: tableData,
}

export const DefaultWithAction = Template.bind({})

DefaultWithAction.args = {
  actionTable: {
    control: action('onClick'),
    field: 'Action',
    label: 'Action',
  },
  data: tableData,
}
