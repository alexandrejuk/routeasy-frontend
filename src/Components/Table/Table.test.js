import React from 'react'
import {
  fireEvent,
  render,
} from '@testing-library/react'
import Table from './'

const mockDelivery = [{
  name: 'Eduardo',
  street: 'R. Ática',
  city: 'São Paulo',
  country: 'Brasil',
  lat: -23.531,
  lng: -46.584,
  weight: 10,
}]

const mockActionTable = {
  control: null,
  field: 'Action',
  label: 'Action',
}

describe('Table component', () => {
  it('should render correctly without action', () => {
    const { container } = render(
      <Table
        data={mockDelivery}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render correctly with action', () => {
    const control = jest.fn()

    const { container, getByTestId } = render(
      <Table
        actionTable={{...mockActionTable, control}}
        data={mockDelivery}
      />
    )
    expect(control).not.toHaveBeenCalled()

    fireEvent.click(getByTestId('button'))

    expect(control).toHaveBeenCalled()
    expect(container).toMatchSnapshot()
  })
})
