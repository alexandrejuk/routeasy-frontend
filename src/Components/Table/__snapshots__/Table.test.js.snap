// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component should render correctly with action 1`] = `
<div>
  <table
    class="table"
  >
    <thead>
      <tr
        class="tr"
      >
        <th
          class="th"
        >
          Nome
        </th>
        <th
          class="th"
        >
          Rua
        </th>
        <th
          class="th"
        >
          Cidade
        </th>
        <th
          class="th"
        >
          País
        </th>
        <th
          class="th"
        >
          Peso
        </th>
        <th
          class="th"
        >
          Lat
        </th>
        <th
          class="th"
        >
          Lng
        </th>
        <th
          class="th"
        >
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="tr"
      >
        <td
          class="td"
        >
          Eduardo
        </td>
        <td
          class="td"
        >
          R. Ática
        </td>
        <td
          class="td"
        >
          São Paulo
        </td>
        <td
          class="td"
        >
          Brasil
        </td>
        <td
          class="td"
        >
          10
        </td>
        <td
          class="td"
        >
          -23.531
        </td>
        <td
          class="td"
        >
          -46.584
        </td>
        <td
          class="td"
        >
          <button
            data-testid="button"
          >
            Action
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table component should render correctly without action 1`] = `
<div>
  <table
    class="table"
  >
    <thead>
      <tr
        class="tr"
      >
        <th
          class="th"
        >
          Nome
        </th>
        <th
          class="th"
        >
          Rua
        </th>
        <th
          class="th"
        >
          Cidade
        </th>
        <th
          class="th"
        >
          País
        </th>
        <th
          class="th"
        >
          Peso
        </th>
        <th
          class="th"
        >
          Lat
        </th>
        <th
          class="th"
        >
          Lng
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="tr"
      >
        <td
          class="td"
        >
          Eduardo
        </td>
        <td
          class="td"
        >
          R. Ática
        </td>
        <td
          class="td"
        >
          São Paulo
        </td>
        <td
          class="td"
        >
          Brasil
        </td>
        <td
          class="td"
        >
          10
        </td>
        <td
          class="td"
        >
          -23.531
        </td>
        <td
          class="td"
        >
          -46.584
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
